config/confighttp/confighttp.go

```diff
@@ -103,8 +104,39 @@ func NewDefaultHTTPClientSettings() HTTPClientSettings {
 	}
 }
 
+// toClientOptions has options that change the behavior of the HTTP client
+// returned by HTTPClientSettings.ToClient().
+type toClientOptions struct {
+	customRoundTripper func(next http.RoundTripper) (http.RoundTripper, error)
+}
+
+// ToClientOption is an option to change the behavior of the HTTP client
+// returned by HTTPClientSettings.ToClient().
+type ToClientOption interface {
+	apply(*toClientOptions)
+}
+
+type toClientOptionFunc func(*toClientOptions)
+
+func (fn toClientOptionFunc) apply(opts *toClientOptions) {
+	fn(opts)
+}
+
+// WithRoundTripper allow for individual components to intercept HTTP requests
+func WithRoundTripper(customRoundTripper func(next http.RoundTripper) (http.RoundTripper, error)) ToClientOption {
```

Comment:
With[Client]RoundTripper to make sure we start a pattern we add Client/Server after with.

Reply:
ping @haoqixu

Reply:
Sorry for the late reply. Please correct me if I've misunderstood.